t/test_postgresql

```diff
@@ -1,16 +1,14 @@
 #!/bin/sh -e
 
+KEEP_DB="${KEEP_DB:-0}"
 DIR=$1
 if test -z "$DIR"; then
     DIR=$(mktemp -d)
 else
+    mkdir -p $DIR
     DIR=$(readlink -f "$DIR")
-fi
-if test -e "$DIR"/postmaster.pid; then
-  pg_ctl -D "$DIR" stop
-fi
-if test -d "$DIR"; then
-  rm -r "$DIR"
+    test -e "$DIR"/postmaster.pid && pg_ctl -D "$DIR" stop
```

Comment:
so if the database exists then now you would always stop it and re-initialize? That's not preserving the behaviour from before.